BayeuxClient.agent.lib.nut

```diff
@@ -0,0 +1,936 @@
+                    _userHandlers[builtChannel](message["channel"], message["data"]);
+                }
+
+                // Example: builtChannel = "/a/b/**"
+                builtChannel += "*";
+            // Example: this branch on the 1-st iteration
+            } else {
+                // Example: builtChannel = "/a/**"
+                builtChannel = channelPart + "/**";
+            }          
+
+            if (builtChannel in _userHandlers) {
+                // TODO: Same as above (imp.wakeup)
+                _userHandlers[builtChannel](message["channel"], message["data"]);
+            }
+        }
+    }
+
+    function _handshake(callback) {
+        local msgId = _nextMessageId();
+        local message = {
+            "id" : msgId,
+            "version" : BAYEUX_CLIENT_PROTOCOL_VERSION,
+            "minimumVersion" : BAYEUX_CLIENT_PROTOCOL_VERSION,
+            "channel" : BAYEUX_CLIENT_META_CHANNEL_HANDSHAKE,
+            "supportedConnectionTypes" : [BAYEUX_CLIENT_LONGPOLL_TRANSPORT_NAME],
+            "advice" : {
+                "timeout" : _advice.timeout,
+                "interval" : _advice.interval
+            }
+        }
+
+        local onResponse = function(response) {
+            _onHandshakeResponse(response, callback);
+        }.bindenv(this);
+
+        _metaHandlers[msgId] <- onResponse;
+
+        local sent = function(error) {
+            if (error != null) {
+                if (msgId in _metaHandlers) {
+                    delete _metaHandlers[msgId];                
+                }
+                callback(error);
+            }
+        }.bindenv(this);
+
+        // We can send it directly without the queue because it should be the first message 
+        // and the other messages are not allowed before the handshake
+        _send(message, sent);
+    }
+
+    function _onHandshakeResponse(response, callback) {
+        _log("Handshake response received:");
+        _logTable(response);
+
+        if (!_checkSuccessfulField(response, callback)) {
+            return;
+        }
+
+        local success = response["successful"];
+
+        if (success && "clientId" in response) {
+            _clientId = response["clientId"];
+            callback(null);
+            return;
+        }
+        local errMsg = null;
+
+        if (success) {
+            // "clientId" NOT in response
+            errMsg = "Handshake response does not have \"clientId\" field!";
+            _logError(errMsg);
+        } else if ("error" in response) {
+            errMsg = response["error"];
+        } else {
+            errMsg = "Handshake was unsuccessful, but response does not have \"error\" field!";
+            _logError(errMsg);
+        }
+        callback(Bayeux.Error(BAYEUX_CLIENT_ERROR_TYPE.BAYEUX_ERROR, errMsg));
+    }
+
+    function _connect(establish = false) {
```

Comment:
what does establish mean?

Reply:
Commented it in the code

Reply:
For some reason, I don't like this construction (connect(establish)) but that's fine as this is a private method.